crates/engine/tree/src/tree/mod.rs

```diff
@@ -673,6 +690,34 @@ where
         Ok(())
     }
 
+    fn on_downloaded_block(&mut self, block: SealedBlockWithSenders) -> Option<TreeEvent> {
+        let block_hash = block.hash();
+        let lowest_buffered_ancestor = self.lowest_buffered_ancestor_or(block_hash);
+        if let Some(status) =
+            self.check_invalid_ancestor_with_head(lowest_buffered_ancestor, block_hash).ok()?
+        {
+            return None
+        }
+
+        if self.is_backfill_active {
+            return None
+        }
+
+        // try to append the block
+        if let Ok(status) = self.insert_block(block) {
+            match status {
+                InsertPayloadOk::Inserted(BlockStatus::Valid(_)) => {
+                    if self.is_sync_target_head(block_hash) {
+                        return Some(TreeEvent::TreeAction(TreeAction::MakeCanonical(block_hash)))
+                    }
+                }
+                _ => return None,
+            }
+        }
+
+        None
```

Comment:
Same here, I think it could be simplified:
```suggestion
        match self.insert_block(block) {
            Ok(InsertPayloadOk::Inserted(BlockStatus::Valid(_))) => {
                if self.is_sync_target_head(block_hash) {
                    return Some(TreeEvent::TreeAction(TreeAction::MakeCanonical(block_hash)))
                }
            }
            _ => return None,
        }
        None
```